src/main/java/build/buildfarm/common/DigestUtil.java

```diff
@@ -71,20 +76,21 @@ public DigestFunction.Value getDigestFunction() {
       return DigestFunction.Value.UNKNOWN;
     }
 
+    // Correct me if I'm wrong but this function doesn't make any sense?
+    // It is always calling the same isValidHexDigest function with the same parameters.
+    // So it will always either return SHA256, or throw an exception.
+    // With the new method we still can't distinguish between SHA256 and BLAKE3, but at least it's better...
```

Comment:
This doesn't seem correct. Yes it calls the same function but on different instance of DigestFunction, so for each call, the value for `hash.bits` will be different. 

You may leave this code as it is, as it doesn't change the code behavior.